li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -325,82 +313,101 @@ public void unsubscribe() {
 
   @Override
   public void commitSync() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitSync(Duration.ofMillis(_defaultApiTimeoutMs));
   }
 
   @Override
   public void commitSync(Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    executeCallbackWithTimeout("commitSync", null,
+        (consumer, ignored, timeoutDuration) -> consumer.commitSync(timeoutDuration), timeout);
   }
 
   @Override
   public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitSync(offsets, Duration.ofMillis(_defaultApiTimeoutMs));
   }
 
   @Override
   public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets, Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+   executeCallbackWithTimeout("commitSync", getPartitionValueMapByCluster(offsets),
+        (consumer, perClusterOffsets, timeoutDuration) -> {
+      consumer.commitSync((Map<TopicPartition, OffsetAndMetadata>) perClusterOffsets, timeoutDuration);
+        }, timeout);
   }
 
   @Override
   public void commitAsync() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitAsync(null);
   }
 
   @Override
   public void commitAsync(OffsetCommitCallback callback) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    for (ClusterConsumerPair<K, V> entry : getImmutableConsumerList()) {
+      entry.getConsumer().commitAsync(callback);
+    }
   }
 
   @Override
   public void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    List<ClusterConsumerPair<K, V>> consumers = getImmutableConsumerList();
+    for (Map.Entry<ClusterDescriptor, Map<TopicPartition, OffsetAndMetadata>> entry :
+        getPartitionValueMapByCluster(offsets).entrySet()) {
+      getOrCreatePerClusterConsumer(consumers, entry.getKey()).commitAsync(entry.getValue(), callback);
```

Comment:
similar to above - user-provided callbacks should be called once

Reply:
Done